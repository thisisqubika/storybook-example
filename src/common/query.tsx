import {
  QueryClient, QueryClientConfig, QueryClientProvider,
} from '@tanstack/react-query';

export function queryClientConfig(
  config?: QueryClientConfig,
): QueryClientConfig {
  return {
    ...config,
    defaultOptions: {
      ...config?.defaultOptions,
      queries: {
        refetchOnWindowFocus: false,
        retry: 1,
        ...config?.defaultOptions?.queries,
      },
    },
  };
} // function newQueryClient

export interface QueryClientWrapperProps {
  config?: QueryClientConfig;
  children: JSX.Element;
} // interface ThemeWrapperProps

export function QueryClientWrapper({
  config,
  children,
}: QueryClientWrapperProps) {
  const queryClient = new QueryClient(queryClientConfig(config));
  return (
    <QueryClientProvider client={queryClient}>
      {children}
    </QueryClientProvider>
  );
} // function QueryClientWrapper

