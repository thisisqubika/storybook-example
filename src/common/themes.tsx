// Check out <https://zenoo.github.io/mui-theme-creator/>.
import {
  createTheme, CssBaseline, ThemeProvider, ThemeOptions,
} from '@mui/material';

export const themeOptions: ThemeOptions = {
  palette: {
    mode: 'light',
    primary: {
      main: '#812020',
    },
    secondary: {
      main: '#cc8333',
    },
    background: {
      default: '#e0cab8',
      paper: '#efe8e2',
    },
    error: {
      main: '#fb3636',
    },
    info: {
      main: '#025e90',
    },
    warning: {
      main: '#e49e01',
    },
    success: {
      main: '#2b9430',
    },
  },
  typography: {
    fontFamily: 'Lora',
    fontSize: 13,
  },
}; // const themeOptions

export interface ThemeWrapperProps {
  children: JSX.Element;
} // interface ThemeWrapperProps

export function ThemeWrapper({
  children,
}: ThemeWrapperProps) {
  return (
    <ThemeProvider theme={createTheme(themeOptions)}>
      <CssBaseline />
      {children}
    </ThemeProvider>
  );
} // function ThemeWrapper
