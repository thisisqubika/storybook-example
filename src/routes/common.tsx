import { isRouteErrorResponse, useRouteError } from 'react-router-dom';

export const PATHS = {
  books: 'books',
}; // const PATHS

export function ErrorBoundary() {
  const error: unknown = useRouteError();
  return isRouteErrorResponse(error) ? (
    <h1>{error.status} {error.statusText}</h1>
  ) : (
    <h1>{(error as Error)?.message ?? error}</h1>
  );
} // function ErrorBoundary
