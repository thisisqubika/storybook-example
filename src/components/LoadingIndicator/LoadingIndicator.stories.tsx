import type { Meta, StoryObj } from '@storybook/react';
import { ThemeWrapper } from '../../common/themes';
import { LoadingIndicator, type LoadingIndicatorProps } from '.';
import { Typography } from '@mui/material';

export default {
  component: LoadingIndicator,
} satisfies Meta<LoadingIndicatorProps>; // meta

export const Base: StoryObj<LoadingIndicatorProps> = {
  args: {
  } satisfies LoadingIndicatorProps,

  render() {
    return (
      <ThemeWrapper>
        <LoadingIndicator />
      </ThemeWrapper>
    );
  },
}; // const Base

export const Error: StoryObj<LoadingIndicatorProps> = {
  args: {
    errorMessage: (
      <Typography variant='h5' color='error'>Errare humanum est!</Typography>
    ),
  } satisfies LoadingIndicatorProps,

  render(props: LoadingIndicatorProps) {
    return (
      <ThemeWrapper>
        <LoadingIndicator {...props} />
      </ThemeWrapper>
    );
  },
}; // const Base
