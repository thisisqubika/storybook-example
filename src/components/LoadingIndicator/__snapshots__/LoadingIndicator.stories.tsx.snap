// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/LoadingIndicator Base smoke-test 1`] = `
<div class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded css-1qbral5-MuiPaper-root"
     data-testid="Loading"
>
  <h5 class="MuiTypography-root MuiTypography-h5 css-1oepjcl-MuiTypography-root">
    Loading, please wait...
  </h5>
  <span class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18li798-MuiCircularProgress-root"
        role="progressbar"
        style="width: 5rem; height: 5rem;"
  >
    <svg class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
         viewbox="22 22 44 44"
    >
      <circle class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
              cx="44"
              cy="44"
              r="20.2"
              fill="none"
              stroke-width="3.6"
      >
      </circle>
    </svg>
  </span>
</div>
`;

exports[`components/LoadingIndicator Error smoke-test 1`] = `
<div class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded css-1qbral5-MuiPaper-root"
     data-testid="Loading"
>
  <h5 class="MuiTypography-root MuiTypography-h5 css-ojno0f-MuiTypography-root">
    Errare humanum est!
  </h5>
  <svg class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeMedium css-1y7yv10-MuiSvgIcon-root"
       focusable="false"
       aria-hidden="true"
       viewbox="0 0 24 24"
       data-testid="ErrorIcon"
  >
    <path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m1 15h-2v-2h2zm0-4h-2V7h2z">
    </path>
  </svg>
</div>
`;
